### ISSUE TITLE
UI

### ISSUE THREAD

**Issue by casperdcl** [tqdm/tqdm#785]
place to keep track of more ways of displaying

- Ubuntu's apt-style progress (eg may fix #784)
- Taskbar icon (e.g. Windows, GNOME, ...)
- Native OS popup (as with file copies)

**Comment by BoPeng** [comment]
@casperdcl Any idea how to achieve the apt-style progress bar? I am willing to work on a PR if it is doable with current tqdm codebase.

**Comment by casperdcl** [comment]
I haven't looked into how it's done but if you can find a python example which can print messages to the bottom of the screen then I can help wrap it quite easily for `tqdm`.

**Comment by BoPeng** [comment]
I think the problem is not sticking the progress bar to the bottom (which can be done through cursor control), but how the rest of the text, of which tqdm has no control, can be displayed properly. `apt` could do it because it has control over both parts but `tqdm` does not have this luxury. 

I mean, the usage pattern would be something like

```
pb = ProgressBar(stick_to_bottom=True)
for i in range(100):
    print(f'some multi-line text for step {i}')
    pb.progress_to(i)
pb.close()
```

The challenge here is that somehow the progressbar should reserve the last row so that when `tqdm` is not in control (after `pb.progress_to(i)`), the `print` statement should not overwrite the progress bar in the last row.

**Comment by casperdcl** [comment]
I assumed that Ubuntu's terminal had some control sequences for the bottom progress bar. If this is not the case then yes, it would likely no be possibly to manage this. The best you sold do would be https://github.com/tqdm/tqdm/#redirecting-writing

**Comment by mikhail** [comment]
+1

"Ubuntu's apt-style progress" would address #945 as well since printing would be above.

My hunch is that this could be possible if tqdm overloaded all streaming handlers (print/sys.stdout/stderr etc) to catch when a new line of text is printed, shift everything up and place the cursor above the top progressbar again.

**Comment by casperdcl** [comment]
yes but I think the issue is `tqdm` needs to be able to work on a range of different systems; so if the implementation only works on Ubuntu's terminal then this would be a candidate for a submodule under `tqdm.contrib` rather than main `tqdm`.
